src/http/server/request_processor.cr

```diff
@@ -19,13 +25,17 @@ class HTTP::Server::RequestProcessor
 
     begin
       until @wants_close
-        request = HTTP::Request.from_io(input)
+        request = HTTP::Request.from_io(
+          input,
+          max_request_line_size: max_request_line_size,
+          max_headers_size: max_headers_size,
+        )
 
         # EOF
         break unless request
 
-        if request.is_a?(HTTP::Request::BadRequest)
-          response.respond_with_error("Bad Request", 400)
+        if request.is_a?(HTTP::Status)
```

Comment:
```suggestion
        if status = request.as?(HTTP::Status)
```

Reply:
I've tried that, unfortunately it doesn't work because it doesn't restrict the type of `request` to `HTTP::Request`.
I figure this should probably work, though because `HTTP::Status` runs into the branch and returns.